components/brave_mobile_subscription/renderer/android/subscription_render_frame_observer.cc

```diff
@@ -23,34 +21,50 @@
 #include "third_party/blink/public/web/web_local_frame.h"
 #include "url/url_util.h"
 
-namespace brave_vpn {
+#if BUILDFLAG(ENABLE_BRAVE_VPN)
+#include "brave/components/brave_vpn/common/brave_vpn_utils.h"
+#endif
+
+namespace brave_subscription {
 
 namespace {
 
-char kIntentParamName[] = "intent";
-char kIntentParamValue[] = "connect-receipt";
-char kIntentParamTestValue[] = "connect-receipt-test";
-char kProductParamName[] = "product";
-char kProductParamValue[] = "vpn";
+inline constexpr char kIntentParamName[] = "intent";
+inline constexpr char kIntentParamValue[] = "connect-receipt";
+inline constexpr char kIntentParamTestValue[] = "connect-receipt-test";
+inline constexpr char kProductParamName[] = "product";
+inline constexpr char kProductVPNParamValue[] = "vpn";
+inline constexpr char kProductLeoParamValue[] = "leo";
 
 }  // namespace
 
-VpnRenderFrameObserver::VpnRenderFrameObserver(
+SubscriptionRenderFrameObserver::SubscriptionRenderFrameObserver(
     content::RenderFrame* render_frame,
     int32_t world_id)
     : RenderFrameObserver(render_frame), world_id_(world_id) {}
 
-VpnRenderFrameObserver::~VpnRenderFrameObserver() = default;
+SubscriptionRenderFrameObserver::~SubscriptionRenderFrameObserver() = default;
 
-bool VpnRenderFrameObserver::EnsureConnected() {
+bool SubscriptionRenderFrameObserver::EnsureConnected() {
+  bool bound = false;
+#if BUILDFLAG(ENABLE_BRAVE_VPN)
   if (!vpn_service_.is_bound()) {
     render_frame()->GetBrowserInterfaceBroker()->GetInterface(
         vpn_service_.BindNewPipeAndPassReceiver());
   }
-  return vpn_service_.is_bound();
+  bound = vpn_service_.is_bound();
+#endif
+#if BUILDFLAG(ENABLE_AI_CHAT)
+  if (!ai_chat_subscription_.is_bound()) {
+    render_frame()->GetBrowserInterfaceBroker()->GetInterface(
+        ai_chat_subscription_.BindNewPipeAndPassReceiver());
+  }
+  bound = ai_chat_subscription_.is_bound();
+#endif
+  return bound;
```

Comment:
this will be false if AI chat feature is disabled and `SubscriptionRenderFrameObserver::DidCreateScriptContext` won't request vpn feature state.

Reply:
yeah, it makes sense, thanks, I corrected it.